src/NativeScript/ObjC/TNSArrayAdapter.mm

```diff
@@ -73,4 +74,13 @@ - (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState*)state objects
     return count;
 }
 
+- (void)dealloc {
+    {
+        JSLockHolder lock(self->_execState);
```

Comment:
Why do we need to lock here but not in `ObjCWrapperObject::~ObjCWrapperObject()`?